src/Service/ProductPositionsUpdater.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ThreeBRS\SortingPlugin\Service;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sylius\Component\Core\Model\ProductTaxonInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use ThreeBRS\SortingPlugin\Service\Exception\ProductPositionsUpdaterException;
+
+class ProductPositionsUpdater
+{
+    private ?FlashBagInterface $sessionBag = null;
+
+    /**
+     * @var array<int|string, int> Key: PK, Value: index
+     */
+    private array $identifiersPositionsMap = [];
+
+    public function __construct(
+        private EntityManagerInterface $entityManager,
+        private RouterInterface $router,
+        private EventDispatcherInterface $eventDispatcher,
+        private TranslatorInterface $translator,
+    )
+    {
+    }
+
+    /**
+     * @throws ProductPositionsUpdaterException
+     */
+    public function process(Request $request): string
+    {
+        $this->initialize($request);
+
+        if ($this->identifiersPositionsMap === []) {
+            $this->setSessionError('threebrs.ui.sortingPlugin.chooseTaxon');
+        }
+
+        $productTaxonCollection = $this->entityManager
+            ->getRepository(ProductTaxonInterface::class)
+            ->findBy(['id' => array_keys($this->identifiersPositionsMap)]);
+
+        if (count($productTaxonCollection) !== count($this->identifiersPositionsMap)) {
+            throw new ProductPositionsUpdaterException('Required identifiers set is invalid.');
+        }
+
+        $taxon = null;
+
+        foreach ($productTaxonCollection as $productTaxon) {
+            assert($productTaxon instanceof ProductTaxonInterface);
+            $taxon ??= $productTaxon->getTaxon();
+            $productTaxon->setPosition($this->identifiersPositionsMap[$productTaxon->getId()] ?? 0);
+        }
+
+        if ($taxon === null) {
+            $this->setSessionError('threebrs.ui.sortingPlugin.noProductMessage');
+
+            return $this->router->generate('threebrs_admin_sorting_index');
```

Comment:
routování vyřešit v konstruktoru

Reply:
V konstruktoru? Asi jsi měl na mysli v kontroleru. Upraveno https://github.com/3BRS/sylius-sorting-plugin/pull/10#discussion_r1196565246

Reply:
jo měl :D